lib/ansible/executor/discovery/python.py

```diff
@@ -0,0 +1,57 @@
+# Copyright: (c) 2018 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# FUTURE: this could be swapped out for our bundled version of distro so long as we maintain Py2.6 compat and
+# don't need to add any other code that would necessitate any kind of script assembly
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import json
+import platform
+import re
+
+
+def get_platform_info():
+    distro = None
+    version = None
+
+    if hasattr(platform, 'dist'):
+        distinfo = platform.dist()
+        distro = distinfo[0]
+        version = distinfo[1]
+
+    if not distro or not version:
+        # try to fall back to parsing /etc/os-release if present
+        osr = parse_os_release()
+        distro = osr.get('ID')
+        version = osr.get('VERSION_ID')
+
+    return dict(distro=distro, version=version)
+
+
+def parse_os_release():
+    contents = dict()
+
+    with open('/etc/os-release', 'rb') as fd:
+        lines = fd.readlines()
+
+    # we only care about a couple of basic values, so this is not a fully-compliant os-release parser
+    linere = re.compile(r'^([\w_]+)="?([\w_\-]+)"?$')
```

Comment:
Two bugs:
* This would need to be a byte string
* This doesn't handle single quoted values.
```suggestion
    linere = re.compile(br'^([\w_]+)=['"]?([\w_\-]+)['"]?$')
```